### DISCUSSION TITLE
Using API to get a list of all feeds and download them one by one

### DISCUSSION THREAD

**Discussion by TheRabidWolverine** [transitland/transitland#428] [Q&A] [answered]
We are trying to get a list of all available feeds, and for all of them, download the latest GTFS dump.

We first tried this API call:

[transit.land/api/v2/rest/feeds?apikey=myfreetrialkey](https://transit.land/api/v2/rest/feeds?apikey=myfreetrialkey)

<img width="440" alt="image" src="https://github.com/transitland/transitland/assets/118490229/d502c1ae-fd97-4ea9-9523-f65192e2b5ca">

It does give us a bunch of operators and feeds, each with an id. However, when we try the feed API with those ids, we can see no feeds in response, all are coming as blank.

[transit.land/api/v2/rest/feeds/12232?apikey=myfreetrialkey](https://transit.land/api/v2/rest/feeds/12232?apikey=GdzEkJQB7bIntulg8pqQlJOaNgByv9Ki)

<img width="174" alt="image" src="https://github.com/transitland/transitland/assets/118490229/dcd02721-b1cc-469e-8ee8-42fec758539d">

<img width="561" alt="image" src="https://github.com/transitland/transitland/assets/118490229/3c72237c-6649-48ce-b3e4-fb83592ed197">

Are we missing something?

**Comment by drewda** [comment, marked as answer]
@TheRabidWolverine in the first screenshot you posted it appears that you are showing the portion of the JSON related to a feed version. So that ID of 12232 is for a feed version, not its parent feed record.

Try looking in the larger JSON object for the ID for the feed. Or alternatively look for the feed Onestop ID, which begins with `f-`
